src/Tasks.UnitTests/AssemblyDependency/ResolveAssemblyReferenceTestFixture.cs

```diff
@@ -861,6 +867,52 @@ internal static bool FileExists(string path)
             return false;
         }
 
+        /// <summary>
+        /// Mock the Directory.GetFiles method.
+        /// </summary>
+        /// <param name="path">The path to directory.</param>
+        /// <returns>'true' if the file is supposed to exist</returns>
+
+        internal static string[] GetDirectoryFiles(string path, string pattern)
+        {
+            if (!Path.IsPathRooted(path))
+            {
+                path = Path.GetFullPath(path);
+            }
+
+            // remove trailing path separator
+            path = Path.GetDirectoryName(Path.Combine(path, "a.txt"));
+
+            if (pattern != "*")
+            {
+                throw new InvalidOperationException("In this context, directory listing with pattern is neither supported not expected to be used.");
+            }
+
+            // Do IO monitoring if needed.
+            if (uniqueGetDirectoryFiles != null)
+            {
+                uniqueGetDirectoryFiles.AddOrUpdate(path, 1, (_, n) => n+1);
+            }
+
+            return s_existentFiles
+                .Where(fn => Path.GetDirectoryName(fn).Equals(path, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should casing here depend on OS?

Reply:
msbuild does not seems to be case sensitive.
In many other places, for example `SystemState.cs` there are many caching dictionaries initialized with StringComparison.OrdinalIgnoreCase. The most closely relevant is `private Dictionary<string, DateTime> instanceLocalLastModifiedCache = new Dictionary<string, DateTime>(StringComparer.OrdinalIgnoreCase);` which is internally used to caching FileExists delegate invocations.